src/py/flwr/common/typing.py

```diff
@@ -208,23 +208,44 @@ class ClientMessage:
 
 
 @dataclass
-class Run:
+class RunStatus:
+    """Run status information."""
+
+    status: str
+    sub_status: str
+    details: str
+
+
+@dataclass
+class Run:  # pylint: disable=too-many-instance-attributes
     """Run details."""
 
     run_id: int
     fab_id: str
     fab_version: str
     fab_hash: str
     override_config: UserConfig
-
-
-@dataclass
-class RunStatus:
-    """Run status information."""
-
-    status: str
-    sub_status: str
-    details: str
+    pending_at: str
+    starting_at: str
+    running_at: str
+    finished_at: str
+    status: RunStatus
+
+    @classmethod
+    def create_empty(cls, run_id: int = 0) -> "Run":
+        """Return an empty Run instance."""
+        return cls(
+            run_id=run_id,
+            fab_id="",
+            fab_version="",
+            fab_hash="",
+            override_config={},
+            pending_at="",
+            starting_at="",
+            running_at="",
+            finished_at="",
+            status=RunStatus(status="", sub_status="", details=""),
```

Comment:
Should we allow creating invalid `Run` objects at all?

Reply:
The purpose of having a `create_empty` method is to simplify the process of creating an empty instance in special cases, such as unit tests. This method isn’t used in our `LinkState`, and in that context, all `Run` objects within `LinkState` are valid. Moreover, since `Run` is a simple dataclass, we can’t guarantee it will always be valid unless we extend it with additional methods, similar to what we’ve done with the `Message` class.

Additionally, we have created "invalid" `Run` objects before, but only for cases where a `Run` instance isn’t available, such as in `grpc-bidi`.

Reply:
Do we use `create_empty` exclusively in tests? If so, we should instead create a helper function in one of the `_test` modules.

Reply:
We also use it in the simulation part.